browser/ui/views/sidebar/sidebar_container_view.h

```diff
@@ -133,17 +153,20 @@ class SidebarContainerView
   void StartBrowserWindowEventMonitoring();
   void StopBrowserWindowEventMonitoring();
 
-  void DoHideSidebar(bool show_event_detect_widget);
-
   raw_ptr<BraveBrowser> browser_ = nullptr;
   raw_ptr<SidePanelCoordinator> side_panel_coordinator_ = nullptr;
   raw_ptr<BraveSidePanel> side_panel_ = nullptr;
   raw_ptr<sidebar::SidebarModel> sidebar_model_ = nullptr;
   raw_ptr<SidebarControlView> sidebar_control_view_ = nullptr;
   bool initialized_ = false;
-  bool is_side_panel_event_ = false;
   bool sidebar_on_left_ = true;
   base::OneShotTimer sidebar_hide_timer_;
+  sidebar::SidebarService::ShowSidebarOption show_sidebar_option_ =
+      sidebar::SidebarService::ShowSidebarOption::kShowAlways;
+  gfx::SlideAnimation show_animation_{this};
+  gfx::SlideAnimation hide_animation_{this};
```

Comment:
I'm wondering why we're using two separate SlideAnimations. Can we use a single SlideAnimation and ::Show(), Hide(), CurrentValue()?

Reply:
Both were used to know which animation is done from end callback. but can know from CurrentValue() also!.
Fixed. thanks.